ntbs-service/Services/CaseManagerSearchService.cs

```diff
@@ -38,8 +38,9 @@ public CaseManagerSearchService(IReferenceDataRepository referenceDataRepository
             // This query is too complex to translate to sql, so we explicitly work on an in-memory list.
             // The size of the directory should make this ok.
             var filtered = caseManagers.Where(c =>
-                    searchKeywords.Any(s => c.FamilyName.ToLower().Contains(s))
-                    || searchKeywords.Any(s => c.GivenName.ToLower().Contains(s))
+                    searchKeywords.Any(s => c.FamilyName != null && c.FamilyName.ToLower().Contains(s))
+                    || searchKeywords.Any(s => c.GivenName != null && c.GivenName.ToLower().Contains(s))
+                    || searchKeywords.Any(s => c.DisplayName.ToLower().Contains(s))
```

Comment:
Would it be worth adding a test for this extra logic? (In particular, a test case where the given name and family name are null)

Reply:
Again this is a case of a service method with no tests for me to start from. I could have a go at writing some tests but it would be quite a bit of work because I don't understand how the pagination works as of yet. What do you think?

Reply:
I think it might be a worthwhile exercise in this case - the service is quite simple, with only one dependency (the repository) that you'll need to mock out. Maybe spend an hour on it and if you haven't been able to get any tests running by then either call me or give up. How does that sound?

Reply:
I've had a look into this, and this runs up against the same issue as [the three unit tests that are currently being skipped](https://github.com/publichealthengland/ntbs_Beta/blob/0a7cc6c0cad904f78dca8f576f4e8503f996023c/ntbs-service-unit-tests/Services/UserServiceTest.cs#L138), basically that `EntityFramework`'s `ToListAsync` extension method doesn't play nice with the normal way you would make a small `IQueryable` list during a test. I think the best solution is to use some more mocking, using [this package](https://www.nuget.org/packages/MockQueryable.Moq/) that is made to solve this specific problem, but that doesn't support .NET Core 2.2 so we would need to wait until Leon finishes the upgrade. 

In short, I reckon I can write some tests for this (and fix the three tests being skipped) but it is going to take some more time, and will soon be blocked behind NTBS-2074. I could just give up and do something else, but then I would need to be given something else to do. What would you like me to do?

Reply:
Right, I understand the problem.

In this case, I think we can do a little better though - there's really no reason for the repository to return a queryable, since this is the only use, and we're immediately enumerating it. So we can do the following:
- Change the signature of the repository method to return a `Task<List<User>>`
- Call `.ToListAsync()` inside the repo
This will then allow us to write the tests that we want without the extra package.

FWIW - if you need a package like the one you've linked to, then you should also question the value of your tests... at that point you're testing logic that will really happen in SQL as though it happens in C# - that's not totally worthless, but it will miss out on a lot. Instead, it might be worth using an in-memory database in your tests (which is a better approximation of using a real database) or writing integration tests (which will actually use a real database).

Reply:
I've added some unit tests, and have the repo method as you have described. Thanks for the advice on testing with entity framework, I've not done lots of it before but what you've said is really helpful and makes a lot of sense

Reply:
The tests look great :-) Thanks!